searchlib/src/vespa/searchlib/features/fieldmatch/computer.cpp

```diff
@@ -15,55 +17,24 @@ using namespace search::fef;
 
 namespace search::features::fieldmatch {
 
-Computer::Computer(const vespalib::string &propertyNamespace, const PhraseSplitter &splitter,
-                   const FieldInfo &fieldInfo, const Params &params) :
-    _splitter(splitter),
-    _fieldId(fieldInfo.id()),
-    _params(params),
-    _useCachedHits(true),
-    _queryTerms(),
-    _queryTermFieldMatch(),
-    _totalTermWeight(0),
-    _totalTermSignificance(0.0f),
-    _fieldLength(FieldPositionsIterator::UNKNOWN_LENGTH),
-    _currentMetrics(this),
-    _finalMetrics(this),
-    _simpleMetrics(params),
-    _segments(),
-    _alternativeSegmentationsTried(0),
-    _cachedHits()
+Computer::Computer(const ComputerSharedState& shared_state, const PhraseSplitter& splitter)
+    : _shared_state(shared_state),
+      _splitter(splitter),
+      _fieldId(_shared_state.get_field_id()),
```

Comment:
Any reason why you copy the fields in the shared state, and not use it where they are ?

Reply:
To keep the data localized while also minimizing the size of the PR diff.